pkg/usersettings/helpers.go

```diff
@@ -1,36 +1,36 @@
 package usersettings
 
 import (
-	"errors"
-
+	authenticationv1 "k8s.io/api/authentication/v1"
 	core "k8s.io/api/core/v1"
 	rbac "k8s.io/api/rbac/v1"
 	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
 )
 
-func newUserSettingMeta(userInfo *unstructured.Unstructured) (*UserSettingMeta, error) {
-	uid := userInfo.GetUID()
-	name := userInfo.GetName()
+func newUserSettingMeta(userInfo authenticationv1.UserInfo) (*UserSettingMeta, error) {
+	uid := userInfo.UID
+	name := userInfo.Username
 	resourceIdentifier := ""
-	ownerReferences := []meta.OwnerReference{}
+	var ownerReferences []meta.OwnerReference
 
 	if uid != "" {
 		resourceIdentifier = string(uid)
 		ownerReferences = []meta.OwnerReference{
 			{
-				APIVersion: userInfo.GetAPIVersion(),
-				Kind:       userInfo.GetKind(),
+				APIVersion: "user.openshift.io/v1", // TODO: is this necessary? What does it do?
+				Kind:       "User",                 // TODO: is this necessary? What does it do?
```

Comment:
This might be out of scope for this PR, but I think we technically don't need theses owner references at all if I understand this usersettings mechanism correctly. Seems like these are only used when creating resources, which are technically not owned by the user. We use the console service account to manage these resources, so shouldn't the onwer references reflect that?

Reply:
Hey @TheRealJon @stlaz, just FYI, the owner reference here exists so that the ConfigMap (and Role and RoleBinding) is automatically removed when the user is removed from a cluster.

This happens esp. often on our Dev Sandbox, where users are removed after 30 days, and it slows down the k8s RBAC checks when we leave all these Role and RoleBindings on the clusters.

Reply:
cc @jhadvig